src/__tests__/transformer.test.ts

```diff
@@ -10,6 +10,29 @@ const stringToNumber = z.string().transform((arg) => parseFloat(arg));
 //   .transform((n) => String(n));
 const asyncNumberToString = z.number().transform(async (n) => String(n));
 
+test("transform ctx.addIssue", () => {
+  const strs = [
+    'foo',
+    'bar'
+  ]
+
+  expect(() => {
+    z
+    .string()
+    .transform((data, ctx) => {
+      const i = strs.indexOf(data)
+      if (i === -1) {
+        ctx.addIssue({
+          code: 'custom',
+          message: `${data} is not one of our allowed strings`,
+        })
+      }
+      return data.length
+    })
+    .parse("asdf");
+  }).toThrow();
```

Comment:
I think it would be better if this test could check for the kind of error, ideally the custom code/message.

Otherwise, this test would also pass if `ctx` is `undefined` as it would throw a JS error

Reply:
Good call. Checking the whole error message now.